src/audio/google/google_ctc_audio_processing_ipc3.c

```diff
@@ -35,7 +35,7 @@ int ctc_set_config(struct processing_module *mod, uint32_t param_id,
 		if (ret)
 			return ret;
 		if (comp_is_new_data_blob_available(cd->tuning_handler)) {
-			comp_get_data_blob(cd->tuning_handler, NULL, NULL);
+			cd->config = comp_get_data_blob(cd->tuning_handler, NULL, NULL);
```

Comment:
now that you have the size, you should be doing a size check from the blob handler against the size in the header. Same for ipc4

Reply:
Thanks for catching this!